core/platform/lf_arduino_support.c

```diff
@@ -134,13 +134,17 @@ int lf_clock_gettime(instant_t* t) {
 */
 int lf_critical_section_enter() { 
     _lf_num_nested_critical_sections++;

```

Comment:
Why are we incrementing twice? Once on this line and once in the if? Also tagging @erlingrj to get more :eyes: on this.

Reply:
That doesnt look right? I dont think `_lf_num_nested_critical_sections++ == 0` will ever be `true` with this design. There are two decrements in the `leave` function also.

Reply:
There were left over decrement and increment statements from when we needed to disable interrupt support, so I reverted it back to what it was before.